pkg/cloudscale/accumulate.go

```diff
@@ -0,0 +1,147 @@
+package cloudscale
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/cloudscale-ch/cloudscale-go-sdk/v2"
+	cloudscalev1 "github.com/vshn/provider-cloudscale/apis/cloudscale/v1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	organizationLabel = "appuio.io/organization"
+	namespaceLabel    = "crossplane.io/claim-namespace"
+)
+
+// AccumulateKey represents one data point ("fact") in the billing database.
+// The actual value for the data point is not present in this type, as this type is just a map key, and the corresponding value is stored as a map value.
+type AccumulateKey struct {
+	Query     string
+	Zone      string
+	Tenant    string
+	Namespace string
+	Start     time.Time
+}
+
+// String returns the full "source" string as used by the appuio-cloud-reporting
+func (k AccumulateKey) String() string {
+	return k.Query + ":" + k.Zone + ":" + k.Tenant + ":" + k.Namespace
+}
+
+/*
+accumulateBucketMetrics gets all the bucket metrics from cloudscale and puts them into a map. The map key is the "AccumulateKey",
+and the value is the raw value of the data returned by cloudscale (e.g. bytes, requests). In order to construct the
+correct AccumulateKey, this function needs to fetch the ObjectUsers's tags, because that's where the zone, tenant and
```

Comment:
Didn't we already move away from the tags and fetch tenant and namespaces information from the cluster? Or better do we still need the tags?

Reply:
true that, that comment is outdated. fixed.